### ISSUE TITLE
Following command line builds don't trigger warnings

### ISSUE THREAD

**Issue by alefranz** [DotNetAnalyzers/StyleCopAnalyzers#3195]
When running dotnet build on an already built project, the StyleCopAnalyzers warnings are not emitted

First build:
```powershell
❯ dotnet build
Microsoft (R) Build Engine version 16.7.0-preview-20360-03+188921e2f for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
Controllers\WeatherForecastController.cs(16,100): warning SA1413: Use trailing comma in multi-line initializers [C:\dev\alefranz\StyleCopDemo\StyleCopDemo\StyleCopDemo.csproj]
Controllers\WeatherForecastController.cs(34,17): warning SA1413: Use trailing comma in multi-line initializers [C:\dev\alefranz\StyleCopDemo\StyleCopDemo\StyleCopDemo.csproj]
  StyleCopDemo -> C:\dev\alefranz\StyleCopDemo\StyleCopDemo\bin\Debug\netcoreapp3.1\StyleCopDemo.dll

Build succeeded.

Controllers\WeatherForecastController.cs(16,100): warning SA1413: Use trailing comma in multi-line initializers [C:\dev\alefranz\StyleCopDemo\StyleCopDemo\StyleCopDemo.csproj]
Controllers\WeatherForecastController.cs(34,17): warning SA1413: Use trailing comma in multi-line initializers [C:\dev\alefranz\StyleCopDemo\StyleCopDemo\StyleCopDemo.csproj]
    2 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.31
```

Following build (without code changes):
```powershell
❯ dotnet build
Microsoft (R) Build Engine version 16.7.0-preview-20360-03+188921e2f for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  StyleCopDemo -> C:\dev\alefranz\StyleCopDemo\StyleCopDemo\bin\Debug\netcoreapp3.1\StyleCopDemo.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.86
```

Repro: https://github.com/alefranz/StyleCopDemo/tree/rebuild-bug

## Expected behaviour

Warnings are always emitted 

## Versions
StyleCop.Analyzers 1.1.118
.NET Core SDK 3.1.401

**Comment by sharwell** [comment]
Duplicate of dotnet/msbuild#3046
